### PR TITLE
Update NUMA to 2.0.14

### PR DESCRIPTION
**Pull request by mkitti**

### CODE REVIEW

File: N/NUMA/build_tarballs.jl [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
 
 name = "NUMA"
-version = v"2.0.13"
+version = v"2.0.14"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/numactl/numactl/releases/download/v2.0.13/numactl-2.0.13.tar.gz",
-                  "991e254b867eb5951a44d2ae0bf1996a8ef0209e026911ef6c3ef4caf6f58c9a"),
+    ArchiveSource("https://github.com/numactl/numactl/releases/download/v2.0.14/numactl-2.0.14.tar.gz",
```
Comment (giordano) [inline N/NUMA/build_tarballs.jl:10]:
```suggestion
    ArchiveSource("https://github.com/numactl/numactl/releases/download/v$(version)/numactl-$(version).tar.gz",
```

**Review by giordano — COMMENTED** [review]
@vchuravy can we require Julia v1.6?

**Comment by vchuravy** [comment]
> @vchuravy can we require Julia v1.6?

Yes!

**Comment by mkitti** [comment]
tests pass with `preferred_gcc_version = v"6"` but not `preferred_gcc_version = v"5"`

File: N/NUMA/build_tarballs.jl [hunk 2]
```diff
@@ -48,4 +48,4 @@ dependencies = [
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"6")
```
Comment (giordano) [inline N/NUMA/build_tarballs.jl:51]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"6", julia_compat="1.6")
```

**Pull request merged**